worker/rhpv2.go

```diff
@@ -287,7 +287,7 @@ func (s *Session) HostKey() types.PublicKey { return s.revision.HostKey() }
 func (s *Session) Read(ctx context.Context, w io.Writer, sections []rhpv2.RPCReadRequestSection, price types.Currency) (err error) {
 	defer wrapErr(&err, "Read")
 	defer recordRPC(ctx, s.transport, s.revision, rhpv2.RPCReadID, &err)()
-	defer recordContractSpending(ctx, s.revision.ID(), api.ContractSpending{Downloads: price}, &err)
+	defer recordContractSpending(ctx, s.revision.ID(), s.revision.Revision.RevisionNumber+1, api.ContractSpending{Downloads: price}, &err)
```

Comment:
`nit` should we pass a pointer to the revision and fetch the id and revision number from the updated revision? It feels a bit weird passing rev+1 here (?)